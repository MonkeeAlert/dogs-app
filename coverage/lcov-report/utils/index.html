
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for utils</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../prettify.css" />
    <link rel="stylesheet" href="../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../index.html">All files</a> utils</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">88.73% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>126/142</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">80.59% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>54/67</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">90.47% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>19/21</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">88.23% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>120/136</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
            <div class="quiet">
                Filter:
                <input oninput="onInput()" type="search" id="fileSearch">
            </div>
        </template>
    </div>
    <div class='status-line high'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="constants.ts"><a href="constants.ts.html">constants.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="8" class="abs high">8/8</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="8" class="abs high">8/8</td>
	</tr>

<tr>
	<td class="file high" data-value="functions.ts"><a href="functions.ts.html">functions.ts</a></td>
	<td data-value="92.62" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 92%"></div><div class="cover-empty" style="width: 8%"></div></div>
	</td>
	<td data-value="92.62" class="pct high">92.62%</td>
	<td data-value="122" class="abs high">113/122</td>
	<td data-value="85.71" class="pct high">85.71%</td>
	<td data-value="63" class="abs high">54/63</td>
	<td data-value="94.73" class="pct high">94.73%</td>
	<td data-value="19" class="abs high">18/19</td>
	<td data-value="92.24" class="pct high">92.24%</td>
	<td data-value="116" class="abs high">107/116</td>
	</tr>

<tr>
	<td class="file low" data-value="tinyUrlHelper.ts"><a href="tinyUrlHelper.ts.html">tinyUrlHelper.ts</a></td>
	<td data-value="41.66" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 41%"></div><div class="cover-empty" style="width: 59%"></div></div>
	</td>
	<td data-value="41.66" class="pct low">41.66%</td>
	<td data-value="12" class="abs low">5/12</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="4" class="abs low">0/4</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="2" class="abs medium">1/2</td>
	<td data-value="41.66" class="pct low">41.66%</td>
	<td data-value="12" class="abs low">5/12</td>
	</tr>

<tr>
	<td class="file empty" data-value="types.ts"><a href="types.ts.html">types.ts</a></td>
	<td data-value="0" class="pic empty">
	<div class="chart"><div class="cover-fill" style="width: 0%"></div><div class="cover-empty" style="width: 100%"></div></div>
	</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
                at Fri Apr 08 2022 23:23:18 GMT+0300 (Eastern European Summer Time)
            </div>
        <script src="../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../sorter.js"></script>
        <script src="../block-navigation.js"></script>
    </body>
</html>
    