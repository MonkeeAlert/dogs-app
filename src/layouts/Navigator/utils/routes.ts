export type RootStackParamList = {
  CatalogTabs: {
    search?: string;
    refresh?: boolean;
  };
  Search: {
    search?: string;
  };
  User: {};
  Gallery: {
    uri: string;
    img: string;
    size: {
      w: number;
      h: number;
    };
    isConnected?: string;
    search?: string;
  };
  Login: {};
};
