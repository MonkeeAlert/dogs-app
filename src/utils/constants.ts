export const colors = {
  white: '#fffafb',
  lightGray: '#E3F1EA',
  turquoise: '#2BD9A5',
  darkTurquoise: '#2E8A6E',
  darkGray: '#2b2c28',
  black: '#131515',
};

export const text = {
  xs: 7,
  s: 14,
  m: 21,
  l: 28,
  xl: 35,
  xxl: 42,
};

export const animationConfig = {
  duration: 200,
};

export const turquoiseGradientArray = [colors.turquoise, '#2bd9a5aa'];
