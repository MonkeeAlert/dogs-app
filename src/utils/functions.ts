export const flatTree = (o: any): string[] => {
  const _o = Object.assign({}, o);
  const keys = Object.keys(_o);

  for (let k in _o) {
    if (_o[k].length > 0) {
      let a = _o[k].map((i: string) => `${k}-${i}`);

      let idx = keys.indexOf(k);

      keys.splice(idx, 1, ...a);
    }
  }

  return keys;
};

export const findBreedInList = (list: string[], breed: string) => {
  const _arr = JSON.parse(JSON.stringify(list));
  const splitted = breed.toLowerCase().split(' ');

  if (
    _arr.indexOf(splitted.join('-')) !== -1 ||
    _arr.indexOf(splitted.reverse().join('-')) !== -1
  ) {
    const normal = _arr.indexOf(splitted.join('-'));
    const reversed = _arr.indexOf(splitted.reverse().join('-'));

    return normal !== -1 ? _arr[normal] : _arr[reversed];
  } else {
    const res = _arr.filter(w => {
      return splitted.filter(i => w.match(i)).length > 0;
    });

    return res.length === 0 ? [] : res.length === 1 ? res[0] : breed;
  }
};

export const parseImage = (s: string) => {
  const fromBreed = s.slice(s.indexOf('breeds'));
  const arr = fromBreed.split('/');

  return `${arr[1]}-${arr[2]}`;
};
