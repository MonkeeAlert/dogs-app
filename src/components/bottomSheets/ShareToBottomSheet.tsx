import {
  View,
  Text,
  StyleSheet,
  FlatList,
  TouchableOpacity,
  Dimensions,
  Pressable,
} from 'react-native';
import React, {ReactElement, useEffect, useRef, useState} from 'react';
import {colors, springConfig, text} from '../../utils/constants';
import {Icon} from 'react-native-elements';
import Animated, {
  Extrapolate,
  interpolate,
  useAnimatedStyle,
  useSharedValue,
  withSpring,
} from 'react-native-reanimated';
import {useImperativeHandle} from 'react';
import {shareImage} from '../../utils/functions';

interface Props {
  children?: ReactElement;
}

const PANEL_HEIGHT = 161;
const ICON_SIZE = 56;
const ICONS = [
  {
    name: 'link-2',
    type: 'feather',
    action: 'CopyLink',
    iconColor: 'light',
    bgColor: colors.darkTurquoise,
  },
  {
    name: 'message-circle',
    type: 'feather',
    iconColor: 'light',
    action: 'MsgApp',
    bgColor: colors.darkTurquoise,
  },
  {
    name: 'mail',
    type: 'feather',
    iconColor: 'light',
    action: 'MailApp',
    bgColor: colors.darkTurquoise,
  },
  // {
  //   name: 'paper-plane',
  //   type: 'ionicon',
  //   iconColor: 'light',
  //   bgColor: colors.darkTurquoise,
  // },
  // {
  //   name: 'facebook',
  //   type: 'feather',
  //   iconColor: 'light',
  //   bgColor: colors.darkTurquoise,
  // },
  // {
  //   name: 'instagram',
  //   type: 'feather',
  //   iconColor: 'light',
  //   bgColor: colors.darkTurquoise,
  // },
];

const ShareToBottomSheet = React.forwardRef(({children}: Props, ref: any) => {
  const [isOpened, setOpened] = useState(false);
  const [uri, setUri] = useState('');

  const handleShareRef = useRef(async (uri: string, type: string) => {
    await shareImage(uri, type);
  });

  const bottomSheet = useSharedValue(0);

  const bgOpacity = useAnimatedStyle(() => ({
    opacity: interpolate(bottomSheet.value, [0, 1], [0, 0.85]),
  }));

  const translation = useAnimatedStyle(() => ({
    transform: [
      {
        translateY: interpolate(
          bottomSheet.value,
          [0, 1],
          [PANEL_HEIGHT, 0],
          Extrapolate.CLAMP,
        ),
      },
    ],
  }));

  useImperativeHandle(ref, () => ({
    toggle: (u: string) => {
      const v = bottomSheet.value === 0 ? 1 : 0;
      bottomSheet.value = withSpring(v, springConfig);

      setUri(u);
      setOpened(prev => !prev);
    },
  }));

  const Content = () => (
    <View style={styles.container}>
      <View style={styles.headerBlock}>
        <Pressable onPress={() => ref.current.toggle()}>
          <Icon
            name={'close-sharp'}
            type={'ionicon'}
            size={21}
            color={colors.darkTurquoise}
            tvParallaxProperties={false}
          />
        </Pressable>
        <Text style={styles.headerText}>Share doggo</Text>
      </View>

      <FlatList
        data={ICONS}
        horizontal={true}
        contentContainerStyle={styles.list}
        keyExtractor={item => `${item.type}_${item.name}`}
        ItemSeparatorComponent={() => <View style={styles.listItemSeparator} />}
        showsHorizontalScrollIndicator={false}
        renderItem={({item}) => (
          <TouchableOpacity
            activeOpacity={0.85}
            onPress={() => handleShareRef.current(uri, item.action)}>
            <View
              style={[styles.listItemIcon, {backgroundColor: item.bgColor}]}>
              <Icon
                type={item.type}
                name={item.name}
                size={28}
                color={item.iconColor === 'light' ? colors.white : colors.black}
                tvParallaxProperties={false}
              />
            </View>
          </TouchableOpacity>
        )}
      />
    </View>
  );

  return (
    <>
      <Animated.View
        ref={ref}
        style={[styles.bottomSheetContainer, translation]}>
        <Content />
      </Animated.View>
      <View
        style={styles.bottomSheetBackgroundContainer}
        pointerEvents={isOpened ? 'auto' : 'none'}>
        <TouchableOpacity
          style={{flex: 1}}
          activeOpacity={1}
          onPress={() => ref.current.toggle()}>
          <Animated.View
            style={[styles.bottomSheetBackgroundInner, bgOpacity]}
          />
        </TouchableOpacity>
      </View>
    </>
  );
});

const styles = StyleSheet.create({
  container: {
    backgroundColor: colors.white,
    paddingBottom: 14,
    paddingTop: 21,
    height: '100%',
    borderTopLeftRadius: 14,
    borderTopRightRadius: 14,
    width: Dimensions.get('screen').width,
  },
  list: {
    marginTop: 21,
    paddingHorizontal: 14,
    flex: 1,
  },
  listItemSeparator: {
    width: 14,
  },
  listItemIcon: {
    width: ICON_SIZE,
    height: ICON_SIZE,
    borderRadius: ICON_SIZE,
    alignItems: 'center',
    justifyContent: 'center',
  },
  headerBlock: {
    flexDirection: 'row',
    justifyContent: 'flex-start',
    alignItems: 'center',
    marginLeft: 14,
  },
  headerText: {
    fontSize: text.m,
    fontWeight: '900',
    color: colors.darkGray,
    marginLeft: 7,
  },
  bottomSheetBackgroundContainer: {
    position: 'absolute',
    top: 0,
    left: 0,
    bottom: 0,
    right: 0,
    width: Dimensions.get('screen').width,
    height: Dimensions.get('screen').height,
  },
  bottomSheetBackgroundInner: {
    flex: 1,
    backgroundColor: colors.black,
  },
  bottomSheetContainer: {
    height: PANEL_HEIGHT,
    zIndex: 1,
    position: 'absolute',
    bottom: 0,
  },
});

export default ShareToBottomSheet;
