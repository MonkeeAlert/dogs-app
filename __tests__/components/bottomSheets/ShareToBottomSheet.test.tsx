import React from 'react';
import {create} from 'react-test-renderer';
import ShareToBottomSheet from '../../../src/components/bottomSheets/ShareToBottomSheet';
import {shareBottomSheetRef} from '../../../src/utils/constants';
import {render, fireEvent, act} from '@testing-library/react-native';

// TODO: pointerEvents issue
describe('ShareToBottomSheet', () => {
  it('matches snapshot', () => {
    const tree = create(
      <ShareToBottomSheet ref={shareBottomSheetRef} />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('shares image on press', () => {
    const {getByTestId} = render(
      <ShareToBottomSheet ref={shareBottomSheetRef} />,
    );
    const btn = getByTestId('ShareToBottomSheet_Content_Link');
    const evt = fireEvent(btn, 'onPress');

    // TODO: rewrite
    // expects Promise
    expect(typeof evt === 'object' && typeof evt.then === 'function').toBe(
      true,
    );
  });

  it('hides bottomsheet when pressing on a specified button', () => {
    const {getByTestId} = render(
      <ShareToBottomSheet ref={shareBottomSheetRef} />,
    );
    const btn = getByTestId('ShareToBottomSheet_CloseBtn');
    const spy = jest.spyOn(shareBottomSheetRef.current, 'toggle');

    act(() => {
      const evt = fireEvent(btn, 'onPress');

      expect(spy).toHaveBeenCalled();
      expect(evt).toBeUndefined();
    });
  });
});
