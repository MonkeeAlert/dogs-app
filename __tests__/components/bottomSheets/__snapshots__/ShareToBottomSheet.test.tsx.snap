// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShareToBottomSheet matches snapshot 1`] = `
Array [
  <View
    animatedStyle={
      Object {
        "value": Object {
          "transform": Array [
            Object {
              "translateY": 177,
            },
          ],
        },
      }
    }
    collapsable={false}
    forwardedRef={
      Object {
        "current": Object {
          "toggle": [Function],
        },
      }
    }
    style={
      Object {
        "bottom": 0,
        "height": 177,
        "position": "absolute",
        "transform": Array [
          Object {
            "translateY": 177,
          },
        ],
        "zIndex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#fffafb",
          "borderTopLeftRadius": 14,
          "borderTopRightRadius": 14,
          "height": "100%",
          "paddingBottom": 14,
          "paddingTop": 21,
          "width": 750,
        }
      }
      testID="ShareToBottomSheet_View"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "marginLeft": 14,
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          testID="ShareToBottomSheet_CloseBtn"
        >
          <View
            style={
              Object {
                "overflow": "hidden",
              }
            }
          >
            <View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "justifyContent": "center",
                  }
                }
              >
                <Text
                  allowFontScaling={false}
                  selectable={false}
                  style={
                    Array [
                      Object {
                        "color": "#2E8A6E",
                        "fontSize": 21,
                      },
                      Object {
                        "backgroundColor": "transparent",
                      },
                      Object {
                        "fontFamily": "Ionicons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                  testID="iconIcon"
                >
                  
                </Text>
              </View>
            </View>
          </View>
        </View>
        <Text
          style={
            Object {
              "color": "#2b2c2c",
              "fontSize": 21,
              "fontWeight": "900",
              "marginLeft": 7,
            }
          }
        >
          Share doggo
        </Text>
      </View>
      <RCTScrollView
        ItemSeparatorComponent={[Function]}
        contentContainerStyle={
          Object {
            "marginTop": 21,
            "paddingHorizontal": 14,
          }
        }
        data={
          Array [
            Object {
              "action": "CopyLink",
              "bgColor": "#2E8A6E",
              "iconColor": "light",
              "name": "link-2",
              "placeholder": "Link",
              "type": "feather",
            },
            Object {
              "action": "MsgApp",
              "bgColor": "#2E8A6E",
              "iconColor": "light",
              "name": "message-circle",
              "placeholder": "Message",
              "type": "feather",
            },
            Object {
              "action": "MailApp",
              "bgColor": "#2E8A6E",
              "iconColor": "light",
              "name": "mail",
              "placeholder": "Mail",
              "type": "feather",
            },
            Object {
              "action": "TelegramApp",
              "bgColor": "#2E8A6E",
              "iconColor": "light",
              "name": "paper-plane",
              "placeholder": "Telegram",
              "type": "ionicon",
            },
            Object {
              "action": "FacebookApp",
              "bgColor": "#2E8A6E",
              "iconColor": "light",
              "name": "facebook",
              "placeholder": "Facebook",
              "type": "feather",
            },
            Object {
              "action": "InstagramApp",
              "bgColor": "#2E8A6E",
              "iconColor": "light",
              "name": "instagram",
              "placeholder": "Instagram",
              "type": "feather",
            },
          ]
        }
        getItem={[Function]}
        getItemCount={[Function]}
        horizontal={true}
        keyExtractor={[Function]}
        onContentSizeChange={[Function]}
        onLayout={[Function]}
        onMomentumScrollBegin={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEventThrottle={50}
        showsHorizontalScrollIndicator={false}
        stickyHeaderIndices={Array []}
        viewabilityConfigCallbackPairs={Array []}
      >
        <View>
          <View
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "flexDirection": "row",
                },
                null,
              ]
            }
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
              testID="ShareToBottomSheet_Content_Link"
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "borderRadius": 56,
                      "height": 56,
                      "justifyContent": "center",
                      "width": 56,
                    },
                    Object {
                      "backgroundColor": "#2E8A6E",
                    },
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "overflow": "hidden",
                    }
                  }
                >
                  <View>
                    <View
                      style={
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "transparent",
                          "justifyContent": "center",
                        }
                      }
                    >
                      <Text
                        allowFontScaling={false}
                        selectable={false}
                        style={
                          Array [
                            Object {
                              "color": "#fffafb",
                              "fontSize": 28,
                            },
                            Object {
                              "backgroundColor": "transparent",
                            },
                            Object {
                              "fontFamily": "Feather",
                              "fontStyle": "normal",
                              "fontWeight": "normal",
                            },
                            Object {},
                          ]
                        }
                        testID="iconIcon"
                      >
                        
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
              <Text
                numberOfLines={2}
                style={
                  Object {
                    "color": "#2b2c2c",
                    "fontSize": 12,
                    "marginTop": 3.5,
                    "textAlign": "center",
                  }
                }
              >
                Link
              </Text>
            </View>
            <View
              style={
                Object {
                  "width": 14,
                }
              }
            />
          </View>
          <View
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "flexDirection": "row",
                },
                null,
              ]
            }
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
              testID="ShareToBottomSheet_Content_Message"
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "borderRadius": 56,
                      "height": 56,
                      "justifyContent": "center",
                      "width": 56,
                    },
                    Object {
                      "backgroundColor": "#2E8A6E",
                    },
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "overflow": "hidden",
                    }
                  }
                >
                  <View>
                    <View
                      style={
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "transparent",
                          "justifyContent": "center",
                        }
                      }
                    >
                      <Text
                        allowFontScaling={false}
                        selectable={false}
                        style={
                          Array [
                            Object {
                              "color": "#fffafb",
                              "fontSize": 28,
                            },
                            Object {
                              "backgroundColor": "transparent",
                            },
                            Object {
                              "fontFamily": "Feather",
                              "fontStyle": "normal",
                              "fontWeight": "normal",
                            },
                            Object {},
                          ]
                        }
                        testID="iconIcon"
                      >
                        
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
              <Text
                numberOfLines={2}
                style={
                  Object {
                    "color": "#2b2c2c",
                    "fontSize": 12,
                    "marginTop": 3.5,
                    "textAlign": "center",
                  }
                }
              >
                Message
              </Text>
            </View>
            <View
              style={
                Object {
                  "width": 14,
                }
              }
            />
          </View>
          <View
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "flexDirection": "row",
                },
                null,
              ]
            }
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
              testID="ShareToBottomSheet_Content_Mail"
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "borderRadius": 56,
                      "height": 56,
                      "justifyContent": "center",
                      "width": 56,
                    },
                    Object {
                      "backgroundColor": "#2E8A6E",
                    },
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "overflow": "hidden",
                    }
                  }
                >
                  <View>
                    <View
                      style={
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "transparent",
                          "justifyContent": "center",
                        }
                      }
                    >
                      <Text
                        allowFontScaling={false}
                        selectable={false}
                        style={
                          Array [
                            Object {
                              "color": "#fffafb",
                              "fontSize": 28,
                            },
                            Object {
                              "backgroundColor": "transparent",
                            },
                            Object {
                              "fontFamily": "Feather",
                              "fontStyle": "normal",
                              "fontWeight": "normal",
                            },
                            Object {},
                          ]
                        }
                        testID="iconIcon"
                      >
                        
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
              <Text
                numberOfLines={2}
                style={
                  Object {
                    "color": "#2b2c2c",
                    "fontSize": 12,
                    "marginTop": 3.5,
                    "textAlign": "center",
                  }
                }
              >
                Mail
              </Text>
            </View>
            <View
              style={
                Object {
                  "width": 14,
                }
              }
            />
          </View>
          <View
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "flexDirection": "row",
                },
                null,
              ]
            }
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
              testID="ShareToBottomSheet_Content_Telegram"
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "borderRadius": 56,
                      "height": 56,
                      "justifyContent": "center",
                      "width": 56,
                    },
                    Object {
                      "backgroundColor": "#2E8A6E",
                    },
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "overflow": "hidden",
                    }
                  }
                >
                  <View>
                    <View
                      style={
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "transparent",
                          "justifyContent": "center",
                        }
                      }
                    >
                      <Text
                        allowFontScaling={false}
                        selectable={false}
                        style={
                          Array [
                            Object {
                              "color": "#fffafb",
                              "fontSize": 28,
                            },
                            Object {
                              "backgroundColor": "transparent",
                            },
                            Object {
                              "fontFamily": "Ionicons",
                              "fontStyle": "normal",
                              "fontWeight": "normal",
                            },
                            Object {},
                          ]
                        }
                        testID="iconIcon"
                      >
                        
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
              <Text
                numberOfLines={2}
                style={
                  Object {
                    "color": "#2b2c2c",
                    "fontSize": 12,
                    "marginTop": 3.5,
                    "textAlign": "center",
                  }
                }
              >
                Telegram
              </Text>
            </View>
            <View
              style={
                Object {
                  "width": 14,
                }
              }
            />
          </View>
          <View
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "flexDirection": "row",
                },
                null,
              ]
            }
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
              testID="ShareToBottomSheet_Content_Facebook"
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "borderRadius": 56,
                      "height": 56,
                      "justifyContent": "center",
                      "width": 56,
                    },
                    Object {
                      "backgroundColor": "#2E8A6E",
                    },
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "overflow": "hidden",
                    }
                  }
                >
                  <View>
                    <View
                      style={
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "transparent",
                          "justifyContent": "center",
                        }
                      }
                    >
                      <Text
                        allowFontScaling={false}
                        selectable={false}
                        style={
                          Array [
                            Object {
                              "color": "#fffafb",
                              "fontSize": 28,
                            },
                            Object {
                              "backgroundColor": "transparent",
                            },
                            Object {
                              "fontFamily": "Feather",
                              "fontStyle": "normal",
                              "fontWeight": "normal",
                            },
                            Object {},
                          ]
                        }
                        testID="iconIcon"
                      >
                        
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
              <Text
                numberOfLines={2}
                style={
                  Object {
                    "color": "#2b2c2c",
                    "fontSize": 12,
                    "marginTop": 3.5,
                    "textAlign": "center",
                  }
                }
              >
                Facebook
              </Text>
            </View>
            <View
              style={
                Object {
                  "width": 14,
                }
              }
            />
          </View>
          <View
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "flexDirection": "row",
                },
                null,
              ]
            }
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
              testID="ShareToBottomSheet_Content_Instagram"
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "borderRadius": 56,
                      "height": 56,
                      "justifyContent": "center",
                      "width": 56,
                    },
                    Object {
                      "backgroundColor": "#2E8A6E",
                    },
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "overflow": "hidden",
                    }
                  }
                >
                  <View>
                    <View
                      style={
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "transparent",
                          "justifyContent": "center",
                        }
                      }
                    >
                      <Text
                        allowFontScaling={false}
                        selectable={false}
                        style={
                          Array [
                            Object {
                              "color": "#fffafb",
                              "fontSize": 28,
                            },
                            Object {
                              "backgroundColor": "transparent",
                            },
                            Object {
                              "fontFamily": "Feather",
                              "fontStyle": "normal",
                              "fontWeight": "normal",
                            },
                            Object {},
                          ]
                        }
                        testID="iconIcon"
                      >
                        
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
              <Text
                numberOfLines={2}
                style={
                  Object {
                    "color": "#2b2c2c",
                    "fontSize": 12,
                    "marginTop": 3.5,
                    "textAlign": "center",
                  }
                }
              >
                Instagram
              </Text>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </View>,
  <View
    pointerEvents="none"
    style={
      Object {
        "bottom": 0,
        "height": 1334,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 750,
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flex": 1,
          "opacity": 1,
        }
      }
      testID="ShareToBottomSheet_OpenBtn"
    >
      <View
        animatedStyle={
          Object {
            "value": Object {
              "opacity": 0,
            },
          }
        }
        collapsable={false}
        style={
          Object {
            "backgroundColor": "#131515",
            "flex": 1,
            "opacity": 0,
          }
        }
      />
    </View>
  </View>,
]
`;
